### ISSUE TITLE
Card ease doesn't appear to be set below 130% until you click "Adjust Ease Factors To Performance"

### ISSUE THREAD

**Issue by buster-blue** [brownbat/autoEaseFactor#24]
In the popup stats when you do a review, the popup says that the cards ease will be set to 1000 (so 100%), but it actually appears to be set to 130% when I checked the card in the browser. After clicking the adjust ease factor button, it properly shows 100% now. I don't know if the card is actually being set to 130%, or if it's really 100% like it should be but just appears to be set to 130%. This happened with every card I checked.

Before I clicked the adjust button:
![Screenshot from 2020-11-10 15-53-16](https://user-images.githubusercontent.com/22484614/98733144-16279680-236e-11eb-80c0-c05829436097.png)
After I clicked the adjust button:
![Screenshot from 2020-11-10 15-53-46](https://user-images.githubusercontent.com/22484614/98733169-20499500-236e-11eb-9280-44b170b31b25.png)

I'm on the Linux version of Anki, and I'm also using the japanese support and awesome TTS addons, but those don't affect scheduling, so I don't think they're affecting anything here. 

Here are my options for the addon:
{
    "leash": 100,
    "max_ease": 5000,
    "min_ease": 1000,
    "moving_average_weight": 0.2,
    "reviews_only": true,
    "stats_duration": 30000,
    "stats_enabled": true,
    "target_ratio": 0.8,
    "two_button_mode": true
}

**Comment by brownbat** [comment]
Sorry for the slow reply! I think I have this one diagnosed. It's a hard fix, but shouldn't actually affect review scheduling.

Testing on Linux Mint with Anki 2.1.35.

For most cards I see it working as designed:

![image](https://user-images.githubusercontent.com/26754/99813359-9213a280-2b15-11eb-8d5a-80d536b8bbde.png)

![image](https://user-images.githubusercontent.com/26754/99813593-de5ee280-2b15-11eb-9cd6-488188ac8e63.png)

However for cards that are just graduating learning status, Anki will reset the ease to a fixed value. It should be determined by "Starting Ease" in deck options. (Is it possible this is only affecting graduating cards for you?)

Graduating a card from a test deck, with deck settings for starting ease at 250:
![image](https://user-images.githubusercontent.com/26754/99814000-6218cf00-2b16-11eb-8052-8ceee7f8a9cf.png)

This is because Anki resets the ease on graduation. Rest easy though, this won't actually affect much, since the next interval (the graduating interval) is set by the deck options, and then after that interval, the algorithm will correct the ease factor to the right value on the very next review and you should be back on track without any mis-scheduling.

It's a little annoying to have that bad data, but shouldn't affect the scheduling of reviews in practice.

**Comment by brownbat** [comment]
Also discussed here:
https://github.com/brownbat/autoEaseFactor/issues/23#issuecomment-719951339

Reopen if you're seeing this on any cards that aren't graduating, but since there's not an easy way for me to stop the graduation process, and since it shouldn't actually impact scheduling, I'm going to close for now.

**Comment by buster-blue** [comment]
This does happen to cards that aren't graduation. The example I showed was actually just a normal review. In fact, I've only seen this on review cards so far, since I haven't had any "new" cards in a while, and I have relearning off completely.

Here's another example. This one shows what the _addon_ says in set the card's ease to:
![Screenshot from 2020-11-20 20-42-31](https://user-images.githubusercontent.com/22484614/99864521-d5036380-2b71-11eb-9040-638659a19c16.png)
Here's what the _browser_ says it set the card to before I clicked the adjust button:
![Screenshot from 2020-11-20 20-44-24](https://user-images.githubusercontent.com/22484614/99864556-0bd97980-2b72-11eb-9f20-7491912fb83b.png)
Here's what the browser says it set the card to after I clicked the adjust button:
![Screenshot from 2020-11-20 20-51-09](https://user-images.githubusercontent.com/22484614/99864595-3c211800-2b72-11eb-9a9b-b8ee23a3caf4.png)

It seems like the addon is saying that it sets the cards to 100% ease, but Anki itself intervenes somehow and raises it to 130% ease, which is the lowest ease that Anki normally allows. Clicking adjust lets the addon change the ease to 100% without Anki's normal scheduling being involved, so it doesn't mess with the addon's scheduling again. At least, that's what I suspect is happening.

**Comment by brownbat** [comment]
Ok, that is really strange.

I want to try to duplicate your conditions as much as possible. 

Can you go into the card browser and select 2-3 cards this has affected (and maybe 2-3 cards this hasn't affected in the same deck for good measure), then export it as an .apkg file (with "include media" and "include scheduling" included?)

If you then just add a .zip extension, GitHub will let you drag it into the comment box as an attachment, like so:
[Selected Notes.apkg.zip](https://github.com/brownbat/autoEaseFactor/files/5577655/Selected.Notes.apkg.zip)

(The shortcut for exporting selected notes in the browser is ctrl+shift+E.)

Thanks for continuing to help me test, sorry for the misdiagnosis. Definitely something weird inserting a floor in there.

**Comment by buster-blue** [comment]
Okay, here are some cards that I saw affected by this issue. The problem seems to happen with any cards that have low enough success rates to be set to 100% ease by the addon, so I actually couldn't find any cards that this _didn't_ affect, since it affected every card that's supposed to be set to 100% ease.

[Selected Notes.apkg.zip](https://github.com/brownbat/autoEaseFactor/files/5578168/Selected.Notes.apkg.zip)

I didn't actually compress the file btw. I just added the extension to the end of the name, so just remove it then import it.

**Comment by brownbat** [comment]
Good catch, can't believe I missed this. 

Anki main is overwriting the factors whenever it gets to a Review or a Lapse.

Reviews:
https://github.com/ankitects/anki/blob/98a4a1927a8e781a1c0b4a297caa8e86134ac027/pylib/anki/schedv2.py#L934

Lapse:
https://github.com/ankitects/anki/blob/98a4a1927a8e781a1c0b4a297caa8e86134ac027/pylib/anki/schedv2.py#L934

v1 lapse:
https://github.com/ankitects/anki/blob/98a4a1927a8e781a1c0b4a297caa8e86134ac027/pylib/anki/sched.py#L515

v1 review:
https://github.com/ankitects/anki/blob/98a4a1927a8e781a1c0b4a297caa8e86134ac027/pylib/anki/sched.py#L554

Since it only affects reviews and lapses, I would see other cards (mid-learning or relearning) go beneath 130% and didn't catch it.

This may be difficult to fix... was hoping not to have to hijack the whole scheduler, for compatibility with future versions and other addons. Might have to though.

**Comment by RisingOrange** [comment]
Would the problem be solved if ease factors were adjusted not (only) during review but outside of it, for example when pressing an "Adjust ease for all decks" button or when closing Anki? I implemented this functionality in PR https://github.com/brownbat/autoEaseFactor/pull/56

**Comment by buster-blue** [comment]
I don't think it would, because it seems that even when the ease factors are manually adjusted outside the review, they just get set back to 130% and the interval increase is based on that. So adjusting outside the review won't help, because the normal anki algorithm interferes when it comes to actually scheduling the next interval, regardless of what the addon sets the ease to.

I actually set the minimum ease to 130% for now (until it gets fixed) because I was afraid that the anki scheduler's interference would give the addon bad data and mess up my review intervals.
